contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,480 @@
+transient slices for resources that are supported by that init system.
 
-- Pod level disk accounting will not be tackled in this proposal.
-- Pod level resource specification in the Kubernetes API will not be tackled in this proposal.
+Depending on the configuration of the associated container runtime,
+operators may have to choose a particular cgroup driver to ensure
+proper system behavior.  For example, if operators use the `systemd`
+cgroup driver provided by the `docker` runtime, the `kubelet` must
+be configured to use the `systemd` cgroup driver.
 
-## Motivations
+Implementation of either driver will delegate to the libcontainer library
+in opencontainers/runc.
 
-Kubernetes currently supports container level isolation only and lets users specify resource requests/limits on the containers [Compute Resources](../../docs/design/resources.md). The `kubelet` creates a cgroup sandbox (via it's container runtime) for each container.
+### Conversion of cgroupfs to systemd naming conventions
 
+Internally, the `kubelet` maintains both an abstract and a concrete name
+for its associated cgroup sandboxes.  The abstract name follows the traditional
+`cgroupfs` style syntax.  The concrete name is the name for how the cgroup
+sandbox actually appears on the host filesystem after any conversions performed
+based on the cgroup driver.
 
-There are a few shortcomings to the current model.
- - Existing QoS support does not apply to pods as a whole. On-going work to support pod level eviction using QoS requires all containers in a pod to belong to the same class. By having pod level cgroups, it is easy to track pod level usage and make eviction decisions.
- - Infrastructure overhead per pod is currently charged to the node. The overhead of setting up and managing the pod sandbox is currently accounted to the node. If the pod sandbox is a bit expensive, like in the case of hyper, having pod level accounting becomes critical.
- - For the docker runtime we have a containerd-shim which is a small library that sits in front of a runtime implementation allowing it to be reparented to init, handle reattach from the caller etc. With pod level cgroups containerd-shim can be charged to the pod instead of the machine.
- - If a container exits, all its anonymous pages (tmpfs) gets accounted to the machine (root). With pod level cgroups, that usage can also be attributed to the pod.
- - Let containers share resources - with pod level limits, a pod with a Burstable container and a BestEffort container is classified as Burstable pod. The BestEffort container is able to consume slack resources not used by the Burstable container, and still be capped by the overall pod level limits.
+If the `systemd` cgroup driver is used, the `kubelet` converts the `cgroupfs`
+style syntax into transient slices, and as a result, it must follow `systemd`
+conventions for path encoding.
 
-## Design
+For example, the cgroup name `/Burstable/pod_123-456` is translated to a
```

Comment:
yes, per our discussion, i think the existing code should following cgroup v2 naming conventions now rather than later.

see: https://www.kernel.org/doc/Documentation/cgroup-v2.txt
section: 2-6-2. Avoid Name Collisions

i will update the text, and open a pr.